compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala

```diff
@@ -329,20 +331,42 @@ object JavaParsers {
     }
 
     def annotations(): List[Tree] = {
-      //var annots = new ListBuffer[Tree]
+      var annots = new ListBuffer[Tree]
       while (in.token == AT) {
         in.nextToken()
-        annotation()
+        annotation() match {
+          case Some(anno) => annots += anno
+          case _ =>
+        }
       }
-      List() // don't pass on annotations for now
+      annots.toList
     }
 
     /** Annotation ::= TypeName [`(` AnnotationArgument {`,` AnnotationArgument} `)`]
       */
-    def annotation(): Unit = {
-      qualId()
-      if (in.token == LPAREN) { skipAhead(); accept(RPAREN) }
-      else if (in.token == LBRACE) { skipAhead(); accept(RBRACE) }
+    def annotation(): Option[Tree] = {
+      val id = convertToTypeId(qualId())
+      var skipAnno = false
+
+      // only parse annotations without arguments
+      if (in.token == LPAREN) {
+        in.nextToken()
+        if (in.token != RPAREN) {
+          skipAnno = true
+          skipAhead(1, 0)
+        }
+        accept(RPAREN)
+      } else if (in.token == LBRACE) {
```

Comment:
In what situation would this occur ? I don't think `@Foo{}` is a valid annotation. If it is somehow, please add a test for it.

Reply:
You are right, `@Foo{}` is not valid. I delete the case.